### PR TITLE
fix golint in dockershim

### PR DESCRIPTION
**Pull request by LexCC**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Fix golint:
pkg/kubelet/dockershim/network/plugins.go:49:2: don't use ALL_CAPS in Go names; use CamelCase
pkg/kubelet/dockershim/network/plugins.go:50:2: don't use ALL_CAPS in Go names; use CamelCase
pkg/kubelet/dockershim/network/testing/plugins_test.go:38:2: don't use underscores in Go names; var all_plugins should be allPlugins
pkg/kubelet/dockershim/network/testing/fake_host.go:50:1: receiver name nh should be consistent with previous receiver name fnh for fakeNetworkHost
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by oomichi** [comment]
@JieJhih Please remove the line `pkg/kubelet/dockershim/network` from `hack/.golint_failures` in this PR to verify all golint failures are fixed in this PR at a CI job.

**Comment by JieJhih** [comment]
@oomichi There still have some `exported method ... should have comment or be unexported` messages when execute `golint ./pkg/kubelet/dockershim/network/`, it seems to me is not all golint failures have been fixed.

**Comment by oomichi** [comment]
It is nice to fix all failures for each directory like the other golint failure fixing PR.

**Comment by JieJhih** [comment]
> It is nice to fix all failures for each directory like the other golint failure fixing PRs.

Sure, I would fix all golint failures in `/pkg/kubelet/dockershim/network/`

File: pkg/kubelet/dockershim/network/testing/fake_host.go [hunk 1]
```diff
@@ -27,28 +27,33 @@ import (
...
-func NewFakeHost(kubeClient clientset.Interface) *fakeNetworkHost {
-	host := &fakeNetworkHost{kubeClient: kubeClient, Legacy: true}
+// NewFakeHost returns a struct that implements the FakeNetworkHost interface
+func NewFakeHost(kubeClient clientset.Interface) *FakeNetworkHost {
+	host := &FakeNetworkHost{kubeClient: kubeClient, Legacy: true}
 	return host
 }
 
-func (fnh *fakeNetworkHost) GetPodByName(name, namespace string) (*v1.Pod, bool) {
+// GetPodByName return empty pod struct
```
Comment (oomichi) [inline pkg/kubelet/dockershim/network/testing/fake_host.go:44]: s/return/returns/

File: pkg/kubelet/dockershim/network/testing/fake_host.go [hunk 2]
```diff
@@ -27,28 +27,33 @@ import (
...
+func (fnh *FakeNetworkHost) GetPodByName(name, namespace string) (*v1.Pod, bool) {
 	return nil, false
 }
 
-func (fnh *fakeNetworkHost) GetKubeClient() clientset.Interface {
+// GetKubeClient return nil for testing
```
Comment (oomichi) [inline pkg/kubelet/dockershim/network/testing/fake_host.go:49]: ditto: s/return/returns/

File: pkg/kubelet/dockershim/network/testing/fake_host.go [hunk 3]
```diff
@@ -27,28 +27,33 @@ import (
...
+func (fnh *FakeNetworkHost) GetKubeClient() clientset.Interface {
 	return nil
 }
 
-func (nh *fakeNetworkHost) SupportsLegacyFeatures() bool {
-	return nh.Legacy
+// SupportsLegacyFeatures return the Legacy value of FakeNetworkHost struct
```
Comment (oomichi) [inline pkg/kubelet/dockershim/network/testing/fake_host.go:54]: ditto: s/return/returns/

File: pkg/kubelet/dockershim/network/testing/fake_host.go [hunk 4]
```diff
@@ -59,10 +64,12 @@ func (nh *fakeNamespaceGetter) GetNetNS(containerID string) (string, error) {
 	return nh.ns, nil
 }
 
+// FakePortMappingGetter record the port mapping for testing
```
Comment (oomichi) [inline pkg/kubelet/dockershim/network/testing/fake_host.go:67]: record?

File: pkg/kubelet/dockershim/network/testing/fake_host.go [hunk 5]
```diff
@@ -59,10 +64,12 @@ func (nh *fakeNamespaceGetter) GetNetNS(containerID string) (string, error) {
...
 type FakePortMappingGetter struct {
 	PortMaps map[string][]*hostport.PortMapping
 }
 
+// GetPodPortMappings return the port by containerID for testing
```
Comment (oomichi) [inline pkg/kubelet/dockershim/network/testing/fake_host.go:72]: ditto: s/return/returns/

File: pkg/kubelet/dockershim/network/testing/mock_network_plugin.go [hunk 6]
```diff
@@ -30,106 +30,127 @@ import (
...
 }
 
-// Recorder for MockNetworkPlugin (not exported)
-type _MockNetworkPluginRecorder struct {
+// MockNetworkPluginRecorder is the recorder for MockNetworkPlugin
+type MockNetworkPluginRecorder struct {
 	mock *MockNetworkPlugin
 }
 
+// NewMockNetworkPlugin return empty MockNetworkPlugin
```
Comment (oomichi) [inline pkg/kubelet/dockershim/network/testing/mock_network_plugin.go:44]: ditto: s/return/returns/

File: pkg/kubelet/dockershim/network/testing/mock_network_plugin.go [hunk 7]
```diff
@@ -30,106 +30,127 @@ import (
...
 func NewMockNetworkPlugin(ctrl *gomock.Controller) *MockNetworkPlugin {
 	mock := &MockNetworkPlugin{ctrl: ctrl}
-	mock.recorder = &_MockNetworkPluginRecorder{mock}
+	mock.recorder = &MockNetworkPluginRecorder{mock}
 	return mock
 }
 
-func (_m *MockNetworkPlugin) EXPECT() *_MockNetworkPluginRecorder {
+// EXPECT return the recorder for MockNetworkPlugin
```
Comment (oomichi) [inline pkg/kubelet/dockershim/network/testing/mock_network_plugin.go:51]: ditto: s/return/returns/

File: pkg/kubelet/dockershim/network/testing/mock_network_plugin.go [hunk 8]
```diff
@@ -30,106 +30,127 @@ import (
...
+func (_m *MockNetworkPlugin) EXPECT() *MockNetworkPluginRecorder {
 	return _m.recorder
 }
 
+// Capabilities return the mock of a set of NET_PLUGIN_CAPABILITY_*
```
Comment (oomichi) [inline pkg/kubelet/dockershim/network/testing/mock_network_plugin.go:56]: ditto: s/return/returns/

File: pkg/kubelet/dockershim/network/testing/mock_network_plugin.go [hunk 9]
```diff
@@ -30,106 +30,127 @@ import (
...
 	return ret0
 }
 
+// Finish checks to see if all the methods that were expected to be called were called
 func (_m *MockNetworkPlugin) Finish() {
 	_m.ctrl.Finish()
 }
 
-func (_mr *_MockNetworkPluginRecorder) Capabilities() *gomock.Call {
+// Capabilities return the mock of a set of NET_PLUGIN_CAPABILITY_* by MockNetworkPluginRecorder
```
Comment (oomichi) [inline pkg/kubelet/dockershim/network/testing/mock_network_plugin.go:68]: ditto: s/return/returns/

**Review by oomichi — COMMENTED** [review]
Thanks for updating. Some small comments.

**Comment by JieJhih** [comment]
> Thanks for updating. Some small comments.

Updated! Please take a look. Thanks!

**Review by oomichi — COMMENTED** [review]
Thanks for updating again.

/triage accepted
/lgtm

File: pkg/kubelet/dockershim/network/testing/mock_network_plugin.go [hunk 10]
```diff
@@ -30,106 +30,127 @@ import (
 	"k8s.io/kubernetes/pkg/kubelet/dockershim/network"
 )
 
-// Mock of NetworkPlugin interface
+// MockNetworkPlugin is the Mock of NetworkPlugin interface
 type MockNetworkPlugin struct {
 	ctrl     *gomock.Controller
-	recorder *_MockNetworkPluginRecorder
+	recorder *MockNetworkPluginRecorder
```
Comment (ehashman) [inline pkg/kubelet/dockershim/network/testing/mock_network_plugin.go:36]: Does this file actually need to be updated? It's generated code, so these changes might end up getting wiped out? The docstrings certainly won't get generated and I think the _-prefaced interface is intentional.

**Comment by SergeyKanzhelev** [comment]
golint is removed from checks https://github.com/kubernetes/kubernetes/pull/98063 and this PR has conflict for a long time. At this stage it fails under the [trivial edits policy](https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#trivial-edits).

/close

**Pull request closed without merging**